### ISSUE TITLE
AMGX.jl support

### ISSUE THREAD

**Issue by termi-official** [SciML/LinearSolve.jl#512]
**Is your feature request related to a problem? Please describe.**

I think it would be nice if we can access [AMGX.jl](https://github.com/JuliaGPU/AMGX.jl) via LinearSolve.jl interface.

**Describe the solution you’d like**

Extension as usual.

**Comment by vboussange** [comment]
We would love to see this feature!

**Comment by AJ0070** [comment]
@ChrisRackauckas is this still wanted, or not needed?

**Comment by ChrisRackauckas** [comment]
Yes ,it would be good to get the integration with this as a callback completed.
